ci/Dockerfile

```diff
@@ -31,80 +35,89 @@ RUN set -eux; \
       --component rustfmt,clippy,rust-src,llvm-tools-preview && \
     rustup default "${RUST_STABLE_VERSION}" && \
     rm -rf "${CARGO_HOME}/registry" "${CARGO_HOME}/git" && \
     # generic ci | "alias" pinned stable toolchain as generic stable
-    ln -s "/usr/local/rustup/toolchains/${RUST_STABLE_VERSION}-x86_64-unknown-linux-gnu" /usr/local/rustup/toolchains/stable-x86_64-unknown-linux-gnu && \
-    # generic ci | install asm toolchain for the default stable toolchain
-    rustup target add riscv64imac-unknown-none-elf && \
-    # needed for `ink-node`
-    rustup target add wasm32v1-none && \
-    cargo install --git https://github.com/use-ink/ink-node --branch main --force --locked && \
+    ln -s "/usr/local/rustup/toolchains/${RUST_STABLE_VERSION}-${RUST_ARCH}-unknown-linux-gnu" "/usr/local/rustup/toolchains/stable-${RUST_ARCH}-unknown-linux-gnu" && \
+    # generic ci | install asm toolchain for the default stable toolchain; `wasm32v1-none` for `ink-node`
+    rustup target add riscv64imac-unknown-none-elf wasm32v1-none && \
     # generic ci | install specific rust nightly, default is stable, use minimum components
     rustup toolchain install "nightly-${RUST_NIGHTLY_VERSION}" --profile minimal \
       --component rustfmt,clippy,rust-analyzer,llvm-tools,rustc-dev,rust-src,llvm-tools-preview && \
-    rustup toolchain install "nightly" --profile minimal \
-      --component rustfmt,clippy,rust-analyzer,llvm-tools,rustc-dev,rust-src,llvm-tools-preview && \
-    # generic ci | "alias" pinned nightly toolchain as generic nightly
-    rustup target add riscv64imac-unknown-none-elf \
-      --toolchain "nightly" && \
-    rustup default nightly && \
-    rustup target add riscv64imac-unknown-none-elf \
-      --toolchain "nightly-${RUST_NIGHTLY_VERSION}" && \
-    rustup run nightly-${RUST_NIGHTLY_VERSION} cargo install cargo-dylint dylint-link && \
-    cargo install cargo-spellcheck --locked && \
-    cargo install cargo-nextest --locked && \
+    # generic ci | "alias" pinned nightly toolchain as rolling nightly
+    ln -s "/usr/local/rustup/toolchains/nightly-${RUST_NIGHTLY_VERSION}-${RUST_ARCH}-unknown-linux-gnu" "/usr/local/rustup/toolchains/nightly-${RUST_ARCH}-unknown-linux-gnu" && \
+    # generic ci | install asm toolchain for the nightly toolchain
+    rustup target add riscv64imac-unknown-none-elf --toolchain "nightly" && \
+    # generic ci | clean up layer
+    rm -rf "${RUSTUP_HOME}/downloads" "${RUSTUP_HOME}/tmp"
+
+# generic ci | install core packages
+RUN cargo +nightly install cargo-dylint dylint-link && \
+    cargo install cargo-spellcheck cargo-nextest xargo --locked && \
     cargo install zepter --locked --version 1.5.1 && \
-    cargo install --git https://github.com/paritytech/cargo-contract \
-      --locked --branch master && \
-    git clone https://github.com/paritytech/polkadot-sdk.git --depth 50 && \
+    cargo install --git https://github.com/use-ink/ink-node --branch main --force --locked && \
```

Comment:
Could be replaced with 'latest' prebuilt ink-node and cargo-contract binaries to speed things up.